### DISCUSSION TITLE
Controlled Switch component does not work properly

### DISCUSSION THREAD

**Discussion by LechK** [twilio-labs/paste#2705] [Q&A] [answered]
Hi. I have noticed, that switch can't be controlled. If we clicking switch, but not changing "on" state, switch still changes.
Easiest way to reproduce, to use it with Alert Dialog.

```
const Example = () => {
  const id = useUID()
  const [on, setOn] = React.useState(false)
  const [alert, setAlert] = React.useState(false)
  return (
    <SwitchContainer id={id} label="I am a controlled Switch">
      <Switch on={on} onClick={(evt) => setAlert(true)} />
    </SwitchContainer>
    <AlertDialog
        heading="Submit application"
        isOpen={alert}
        onConfirm={setOn(!on)}
        onConfirmLabel="Submit"
        onDismiss={setAlert(false)}
        onDismissLabel="Cancel"
      >
       Alert something
    </AlertDialog>
  )
}
```
Because of that, we can't use Switch in some situations.

**Comment by ghost** [comment]
Thanks for the discussion here @LechK. I looked at the implementation and this does look like a bug! Our component doesn't take into account the presence of an `onClick` callback to manage controlled state, it also doesn't provide a similar check for keyDown events. I'll share with the team and we'll work on a fix

**Reply by TheSisb** [reply]
Hi @LechK,

Small update that we're actively working on a fix to this component. Hoping to have it ready by next week at the latest.

**Reply by nkrantz** [reply]
Here's the PR if you'd like to follow along: https://github.com/twilio-labs/paste/pull/2753

**Comment by shleewhite** [comment, marked as answer]
@LechK The fix has merged and will be available in our next version of the package: https://github.com/twilio-labs/paste/pull/2779.

**Comment by LechK** [comment]
ThankYou!

2022-10-28, pn, 19:48 Lee White ***@***.***> rašė:
